service/src/main/kotlin/fi/espoo/evaka/assistanceneed/preschooldecision/AssistanceNeedPreschoolDecisionService.kt

```diff
@@ -53,38 +52,77 @@ class AssistanceNeedPreschoolDecisionService(
     private val documentClient: DocumentService,
     private val templateProvider: ITemplateProvider,
     private val messageProvider: IMessageProvider,
-    private val sfiClient: SfiMessagesClient,
     bucketEnv: BucketEnv,
     private val emailEnv: EmailEnv,
-    asyncJobRunner: AsyncJobRunner<AsyncJob>
+    private val asyncJobRunner: AsyncJobRunner<AsyncJob>
 ) {
     private val bucket = bucketEnv.data
 
     init {
-        asyncJobRunner.registerHandler(::runSendAssistanceNeedPreschoolDecisionEmail)
         asyncJobRunner.registerHandler(::runCreateAssistanceNeedPreschoolDecisionPdf)
+        asyncJobRunner.registerHandler(::runSendAssistanceNeedPreschoolDecisionEmail)
         asyncJobRunner.registerHandler(::runSendSfiDecision)
     }
 
-    fun runSendAssistanceNeedPreschoolDecisionEmail(
+    fun runCreateAssistanceNeedPreschoolDecisionPdf(
         db: Database.Connection,
         clock: EvakaClock,
-        msg: AsyncJob.SendAssistanceNeedPreschoolDecisionEmail
+        msg: AsyncJob.CreateAssistanceNeedPreschoolDecisionPdf
     ) {
-        this.sendDecisionEmail(db, msg.decisionId, clock.today())
-        logger.info {
-            "Successfully sent assistance need preschool decision email (id: ${msg.decisionId})."
+        val decisionId = msg.decisionId
+        db.transaction { tx ->
+            val decision = tx.getAssistanceNeedPreschoolDecisionById(decisionId)
+            val endDate =
+                tx.getAssistanceNeedPreschoolDecisionsByChildId(decision.child.id)
+                    .find { it.id == decisionId }
+                    ?.validTo
+
+            check(!decision.hasDocument) {
+                "Assistance need preschool decision $decisionId already has a document key"
```

Comment:
Would the be more understandable explanation: "A pdf already exists for the assistance need preschool decision @decisionId"